File: langhuan-service-java/src/main/java/com/langhuan/utils/rag/metadata/RagMetadataFactory.java
package com.langhuan.utils.rag.metadata;

import com.langhuan.model.domain.TRagFile;
import org.springframework.stereotype.Component;

import java.util.Map;

/**
 * @author Afish
 * @date 2025/7/23 13:51
 */
@Component
public class RagMetadataFactory {
    public Map<String, Object> createMetadata(TRagFile ragFile) {
        return Map.of(
                "filename", ragFile.getFileName(),
                "fileId", String.valueOf(ragFile.getId()),
                "groupId", ragFile.getFileGroupId(),
                "rank", 0
        );
    }

    public Map<String, Object> createMetadata(String fileName, String fileId, String groupId) {
        return Map.of(
                "filename", fileName,
                "fileId", fileId,
                "groupId", groupId,
                "rank", 0
        );
    }
}

File: langhuan-server/src/main/kotlin/com/langhuan/utils/rag/metadata/RagMetadataFactory.kt
package com.langhuan.utils.rag.metadata

import com.langhuan.model.domain.TRagFile
import org.springframework.stereotype.Component

/**
 * @author Afish
 * @date 2025/7/23 13:51
 */
@Component
class RagMetadataFactory {
    fun createMetadata(ragFile: TRagFile): Map<String, Any?> {
        return mapOf(
                "filename" to ragFile.fileName,
                "fileId" to ragFile.id.toString(),
                "groupId" to ragFile.fileGroupId,
                "rank" to 0
        )
    }

    fun createMetadata(fileName: String, fileId: String, groupId: String): Map<String, Any> {
        return mapOf(
                "filename" to fileName,
                "fileId" to fileId,
                "groupId" to groupId,
                "rank" to 0
        )
    }
}
